### PR TITLE
add drag and drop

### PR DESCRIPTION
**Pull request by BlackAnge**

### CODE REVIEW

File: app/javascript/components/Tasks/index.js [hunk 1]
```diff
@@ -126,42 +133,68 @@ class Tasks extends React.Component {
...
+  onDragEnd = (result) => {
+    if (isEmpty(result.destination)) return
+
+    const tasks = this.reorder(
+      this.state.tasks,
+      result.source.index,
+      result.destination.index
+    )
+
+    const data = tasks.map((task) => { return { klass: 'Task', id: task.id } })
```
Comment (0bman) [inline app/javascript/components/Tasks/index.js]: `const data = tasks.map((task) => ({ klass: 'Task', id: task.id }))`

File: app/javascript/components/Tasks/index.js [hunk 2]
```diff
@@ -126,42 +133,68 @@ class Tasks extends React.Component {
...
+
+    this.setState({ tasks })
+
+    api.postSort('sortable/reorder', data)
```
Comment (0bman) [inline app/javascript/components/Tasks/index.js]: `{ rails_sortable: body }`

File: app/javascript/components/Tasks/index.js [hunk 3]
```diff
@@ -42,6 +43,13 @@ class Tasks extends React.Component {
     return filteredArray.filter(task => task.status === params)
   }
 
+  reorder = (list, startIndex, endIndex) => {
```
Comment (0bman) [inline app/javascript/components/Tasks/index.js]: update this method

File: app/javascript/components/Tasks/index.js [hunk 4]
```diff
@@ -42,6 +43,13 @@ class Tasks extends React.Component {
...
+    const result = [...list]
+    const [removed] = result.splice(startIndex, 1)
```
Comment (BlackAnge) [inline app/javascript/components/Tasks/index.js]: this const return single object, without `[...]`, return mass of objects

File: app/serializers/api/v1/task_serializer.rb [hunk 5]
```diff
@@ -2,4 +2,8 @@
 
 class Api::V1::TaskSerializer < ActiveModel::Serializer
   attributes :id, :title, :completed_to, :completed_at, :status
+
+  def completed_at
+    Time.now.localtime&.strftime('%e %b %Y %H:%M') if object.completed_at
```
Comment (0bman) [inline app/serializers/api/v1/task_serializer.rb]: 💩

**Pull request merged**